src/turbolinks/history.coffee

```diff
@@ -35,11 +30,18 @@ class Turbolinks.History
         restorationIdentifier = turbolinks.restorationIdentifier
         @delegate.historyPoppedToLocationWithRestorationIdentifier(location, restorationIdentifier)
 
+  onPageLoad: (event) =>
+    Turbolinks.defer =>
+      @pageLoaded = true
+
   # Private
 
   shouldHandlePopState: ->
     # Safari dispatches a popstate event after window's load event, ignore it
-    pageLoaded is true
+    @pageIsLoaded()
+
+  pageIsLoaded: ->
+    @pageLoaded or document.readyState is "complete"
```

Comment:
👍 Similar to [what JQuery does](https://github.com/jquery/jquery/blob/master/src/core/ready.js#L80).